.github/workflows/dns-apply.yml

```diff
@@ -0,0 +1,30 @@
+---
+name: Apply DNS changes
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - dns/
```

Comment:
Would this give me (a member of non-critical infra) power to deploy DNS changes (because I have the power to merge PRs)? Or would I not have the power to merge it because CODEOWNERS mentions `@NixOS/infra-build`? I suppose I could get my hands on the `secrets.NETLIFY_TOKEN` if I was being malicious.

Reply:
It would be possible, but it would be also visible if you do it because, you cannot force push the malicious code afterwards.

Reply:
But I have a look if github's OIDC could be used to reduce the TTL of these token drastically.

Reply:
Doesn't look like external OIDC providers are supported by netlify.

Reply:
(Even if you could force push [do we not have branch protection set up here? {can infra-build members even override that?}], they're persisted in the audit log and [in the UI](https://github.com/NixOS/infra/activity) (which I didn't know about till recently!))

Reply:
We require codeowner approval for merges (or repo admin bypass), so all DNS changes are controlled by in practice controlled by infra-build.

Reply:
> We require codeowner approval for merges (or repo admin bypass), so all DNS changes are controlled by in practice controlled by infra-build.

A theoretical evil version of @jfly could make a new branch here with a modified workflow that deploys any changes or leaks the Netlify token without codeowner approval (or even a PR), though, right?